library/src/main/kotlin/com/livefront/gsonkotlinadapter/util/KParameter.kt

```diff
@@ -3,8 +3,30 @@ package com.livefront.gsonkotlinadapter.util
 import com.google.gson.annotations.SerializedName
 import java.lang.reflect.Field
 import java.lang.reflect.Modifier
+import kotlin.reflect.KClass
 import kotlin.reflect.KParameter
 
+/**
+ * Returns the default value of the primitive this [KParameter] represents or `null` if it does not
+ * represents primitive.
+ */
+internal val KParameter.defaultPrimitiveValue: Any?
+    get() {
+        if (type.isMarkedNullable) return null
+        return when (type.classifier as? KClass<*>) {
+            null -> null
+            Boolean::class -> false
+            Byte::class -> 0.toByte()
+            Char::class -> '\u0000'
```

Comment:
👍 

It looks like `'\u0000'` is equal to `0.toChar()` as well.

Reply:
Oh interesting...that might be worth trying.